SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py

```diff
@@ -219,24 +219,27 @@ def data_frame_ac9_hs6(self):
 
     def data_frame_absorption(self):
         """ return data as a pandas data frame """
+        col0_data   = self.sheet.column_at(0)[self.idx_start_data:]  # column zero starting at DATA part
+        idx_col_val = range(2, self.max_data_col + 2)  # + 2 relates to 2 empty cols before start of data. range of data col idx
+
         n_var_rows_defined = self.idx_end_var_rows_def - self.idx_start_var_rows_def - 1
         data = []
         for i in range(self.idx_start_data + n_var_rows_defined + 1, self.idx_end_data):
-            data.append(self.sheet.row_at(i)[2:self.max_data_col + 2])
+            data.append([self.sheet.row_at(i)[j] for j in idx_col_val])
         data_frame = pd.DataFrame(data)
 
         dates                 = self.sheet.row_at(self.idx_start_data + 1)[2:self.max_data_col + 2]
         data_frame.Dates      = pd.to_datetime(dates)
         data_frame.Wavelength = self.sheet.column_at(1)[self.idx_start_data + n_var_rows_defined + 1:]
 
-        data_frame.Station_Code = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Station_Code'))[2:self.max_data_col + 2]
-        data_frame.Latitude     = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Latitude'))[2:self.max_data_col + 2]
-        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Longitude'))[2:self.max_data_col + 2]
-        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Depth'))[2:self.max_data_col + 2]
+        data_frame.Station_Code = [self.sheet.row_at(self.idx_start_data + col0_data.index('Station_Code'))[i] for i in idx_col_val]
+        data_frame.Latitude     = [self.sheet.row_at(self.idx_start_data + col0_data.index('Latitude'))[i] for i in idx_col_val]
+        data_frame.Longitude    = [self.sheet.row_at(self.idx_start_data + col0_data.index('Longitude'))[i] for i in idx_col_val]
+        data_frame.Depth        = [self.sheet.row_at(self.idx_start_data + col0_data.index('Depth'))[i] for i in idx_col_val]
```

Comment:
`col0_data` definitely helps, but for me this list comprehension with `idx_col_val` is neither shorter nor clearer, so I'm not sure why you would do it.